src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -250,6 +258,76 @@ private EmailWrapper generateSubmissionConfirmationEmail(
 
     }
 
+    private EmailWrapper generateLinkRecoveryEmailForUser(String userEmail) {
+        Instant endTime = Instant.now();
+        Instant startTime = endTime.minus(Duration.ofDays(180));
+        String subject = EmailType.FEEDBACK_ACCESS_LINKS_RESENT.getSubject();
+        StringBuffer linksFragmentValue = new StringBuffer(1000);
```

Comment:
why not StringBuilder?

Reply:
and 1000 might be too low a limit? If the user has a lot of sessions within the past 180 days, then it might be an issue...